network/p2p/network.go

```diff
@@ -153,24 +151,22 @@ func (n *Network) Disconnected(_ context.Context, nodeID ids.NodeID) error {
 
 // NewClient returns a Client that can be used to send messages for the
 // corresponding protocol.
-func (n *Network) NewClient(self ids.NodeID, handlerID uint64, options ...ClientOption) *Client {
-	client := &Client{
-		self:          self,
+func (n *Network) NewClient(handlerID uint64, options ...ClientOption) *Client {
+	clientOptions := &clientOptions{}
+
+	for _, option := range options {
+		option.apply(clientOptions)
+	}
+
+	return &Client{
 		handlerID:     handlerID,
 		handlerIDStr:  strconv.FormatUint(handlerID, 10),
 		handlerPrefix: ProtocolPrefix(handlerID),
 		sender:        n.sender,
 		router:        n.router,
-		options: &clientOptions{
-			nodeSampler: n.Peers,
-		},
+		options:       clientOptions,
+		sampler:       newPeerSampler(n.Peers, clientOptions.samplingFilters...),
```

Comment:
While we don't currently use it, it feels like it may behoove us to support sampling by stake weight as well (this only really supports sampling uniformly).